submissions/iakow/a-tiny-js-work/index.js

```diff
@@ -0,0 +1,63 @@
+const man = {
+   species: 'human',
+   name: 'Donald',
+   gender: 'male',
+   legs: 2,
+   hands: 2,
+   saying: 'Aloha!',
+};
+
+const woman = {
+   species: 'human',
+   name: 'Eva',
+   gender: 'female',
+   legs: 2,
+   hands: 2,
+   saying: 'Hello',
+};
+
+const cat = {
+   species: 'cat',
+   name: 'Murka',
+   gender: 'female',
+   legs: 4,
+   hands: 0,
+   saying: 'Meow...',
+};
+
+const dog = {
+   species: 'dog',
+   name: 'Gektor',
+   gender: 'male',
+   legs: 4,
+   hands: 0,
+   saying: 'WOFFF!',
+};
+
+const catWoman = {
+   species: 'humacat',
+   name: 'Akrum',
+   gender: 'female',
+   legs: 2,
+   hands: 2,
+   saying: cat.saying,
+};
+
+man.friends = [woman, dog];
+woman.friends = [man, cat];
+dog.friends = [man, woman, catWoman, cat];
+catWoman.friends = [cat, woman];
+
+const printInhabitant = obj => {
+   const props = Object.keys(obj).map(key => {
+      if (key === 'friends') {
+         return obj[key].map(friend => friend.name).join(', ');
+      }
+
+      return obj[key];
+   })
+
+   print(props.join('; '));
+}
+
+[man, woman, cat, dog, catWoman].forEach(printInhabitant);
```

Comment:
`.forEach(printInhabitant)` is a nice shortcut if compared with `.forEach(obj => printInhabitant(obj))`
You might keep it that concise if you introduce `toString` method to every object (effectively bounding `printInhabitant` function to `toString` object method).
This, however will affect they way you enumerate object's own properties.

**You do not need to implement this now.** You will have a chance yet after the subsequent task.

Getting familiar with [a built-in `toString` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString) will be sufficient at this stage (check **Examples** section).

Reply:
Thanks. Looked at examples and I have one question, if it's convenient for you. Why is a named function rather than an anonymous one used when overriding toString()?

Reply:
Good question. Have no idea why they used a named function in the examples.
The only valid, to my knowledge, use case of a named function on the right side of an assignment is when that function needs to call itself recursively.
E.g. `const func = function rfunc(n){if (n=0) return; rfunc(n-1);};`
`func` is not defined yet and one needs a defined name to call the function from inside itself.

Reply:
Thanks for the answer!